crates/apollo-parser/src/parser/grammar/fragment.rs

```diff
@@ -51,9 +51,14 @@ pub(crate) fn type_condition(p: &mut Parser) {
             if p.peek_data().unwrap() == "on" {
                 p.bump(SyntaxKind::on_KW);
             } else {
-                p.err("exptected 'on'");
+                p.err("expected 'on'");
+            }
+
+            if let Some(TokenKind::Name) = p.peek() {
+                ty::named_type(p)
+            } else {
+                p.err("expected a Name")
```

Comment:
a bit more precise, since so many other things are a Name in graphql grammar
```suggestion
                p.err("expected a Name Type in a Type Condition")
```